Anthony 1452 add gsd mode to sidebar

anthony-hull opened the pull request:
### Details
This PR does two things:
Implements a new picker and also data manipulation in the data layer.

### New Picker
Adding this library required me to make some changes to WebPack and Babel. 
I needed babel to transpile the flow annotations so I added the [babel preset-flow](https://babeljs.io/docs/en/v7-migration#separation-between-the-react-and-flow-presets)
Additionally I needed to add these libraries to the exclude pattern to get the project to build:

- react-native-picker-select
- react-native-web
- @react-native-picker

I also renamed one of the plugins to use [the new name space](https://babeljs.io/docs/en/v7-migration#package-renames)

More reading:
https://github.com/babel/babel/issues/8379

I also had to fix an error with the picker input dependancy by adding @react-native-picker/picker as a named dependancy in the project.

### Datalayer
This was the easier part, there is a new test for the change in logic reached by the GSD flag.

### Fixed Issues

Fixes https://github.com/Expensify/Expensify.cash/issues/1452

### Tests
Run the automated unit tests for the data driven test.
E2E tests requires you to have a few chats that have a mixture of pinned, unpinned and read and unread reports.
You can then open the settings panel and use the drop down to change the chat priority mode between GSD and the default.

  

### Tested On

- [x] Web
- [x] Mobile Web
- [x] Desktop
- [x] iOS
- [x] Android

### Screenshots

#### Web
![web GSD screenshot](https://user-images.githubusercontent.com/47184118/108928531-3a1b9900-763a-11eb-8aad-c8c7f0f2d8dd.png)
#### Mobile Web
![mobile web GSD screenshot](https://user-images.githubusercontent.com/47184118/108928614-5ddedf00-763a-11eb-9d24-78f1c01b814b.png)
#### Desktop
![desktop GSD screenshot](https://user-images.githubusercontent.com/47184118/108928511-31c35e00-763a-11eb-996e-9900ebaf0b71.png)
#### iOS
![iOS GSD screenshot open modal](https://user-images.githubusercontent.com/47184118/108928534-3b4cc600-763a-11eb-94d1-b8a11f4755f9.png)
![iOS GSD screenshot](https://user-images.githubusercontent.com/47184118/108928536-3be55c80-763a-11eb-849b-2ccd19870f07.png)

#### Android
![image](https://user-images.githubusercontent.com/47184118/108928670-751dcc80-763a-11eb-90cd-971ca46de1c1.png)
![image](https://user-images.githubusercontent.com/47184118/108928696-81a22500-763a-11eb-8551-3c0b26eb5ad5.png)

anthony-hull committed:
add picker and default chat priority mode copy + styling

anthony-hull committed:
Added tests for new GSD mode

also remove bad CSS

anthony-hull committed:
GSD logic added.

setSidebarOptions updated with logic to control sideBarOptions filter and sort parameters
SettingsPage.js gets priority mode tag from Onyx

anthony-hull committed:
WIP refactoring to picker select

anthony-hull committed:
added RNPickerSelect + WebPack and Babel changes to enable

anthony-hull committed:
Add styling to all platforms refactor data props

anthony-hull commented:
@marcaaron 
Hi Marc, since you had the most input in the issue, I think you would be best suited to review this PR

anthony-hull commented on assets/images/down.svg:
```diff
@@ -0,0 +1,6 @@
+<svg version="1.1" id="Layer_1" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" x="0px" y="0px"
+	 viewBox="0 0 20 20" style="enable-background:new 0 0 20 20;" xml:space="preserve">
+<path d="M 17.942 7.325 L 11.342 13.925 C 10.542 14.725 9.442 14.725 8.742 13.925 L 2.142 7.325 C 1.942 7.125 1.942 6.725 2.142 6.425 L 2.942 5.625 C 3.142 5.425 3.542 5.425 3.842 5.625 L 9.942 11.725 L 16.042 5.625 C 16.242 5.425 16.642 5.425 16.942 5.625 L 17.842 6.525 C 18.042 6.725 18.042 7.125 17.942 7.325 Z"/>
+</svg>
```
annoyingly I couldn't pass a transform property into the SVG component as it crashed the RNSVG viewer on android. So I had to edit the existing SVG and rotate it

marcaaron replied:
This seems fine. I'm not sure using a transform would be worth the added complexity anyway.

anthony-hull commented on babel.config.js:
```diff
@@ -1,9 +1,9 @@
     ['react-native-web'],
 
-    '@babel/transform-runtime',
+    '@babel/plugin-transform-runtime',
```
https://babeljs.io/docs/en/v7-migration#package-renames

marcaaron replied:
Can you elaborate on this change? Not sure I see why it's necessary.

Julesssss replied:
+1. I read through the [linked issue](https://github.com/babel/babel/issues/8379) you mention in the description, but I'm still not sure what the original issue was here that requires the Babel update.

anthony-hull replied:
It's not necessary. I'll revert. I renamed it to the the whole name instead of the shorthand because on that link above it implied they (the maintainers) preferred full names for clarity to the shorthand.

anthony-hull commented on babel.config.js:
```diff
@@ -1,9 +1,9 @@
-const defaultPresets = ['@babel/preset-react', '@babel/preset-env'];
+const defaultPresets = ['@babel/preset-react', '@babel/preset-env', '@babel/preset-flow'];
```
I needed babel to transpile the flow annotations so I added the babel preset-flow

marcaaron replied:
Makes sense to me.

anthony-hull commented on src/libs/OptionsListUtils.js:
```diff
@@ -368,18 +376,37 @@ function getNewGroupOptions(
+            };
+            break;
+        default:
+            break;
```
This is maybe a premature generalisation, but I built it like that to allow easy extending to include more priority modes/option combinations.

marcaaron replied:
>This is maybe a premature generalisation

I'd say it is. Let's please just `check priorityMode === 'gsd'`.

Julesssss replied:
+1

anthony-hull commented on src/libs/actions/Account.js:
```diff
@@ -27,7 +27,7 @@ function fetchPriorityMode() {
 }
 
-export default {
+export {
```
the other action files didn't use exports I changed this to be consistent with the others.

marcaaron replied:
Would you mind undoing this? It seems a little unrelated to the task at hand.

anthony-hull commented on src/pages/SettingsPage.js:
```diff
@@ -68,7 +94,7 @@ const SettingsPage = ({
             >
                 <View style={styles.settingsWrapper}>
                     <View
-                        style={[styles.largeAvatar, styles.mb3]}
```
this was a typo, it doesn't exist. There is a similar class name to this but it already gets applied within the component

anthony-hull commented on src/styles/styles.js:
```diff
@@ -169,6 +169,43 @@ const styles = {
+            right: 12,
+        },
+    },
+
```
I know this is a different structure to the other styled. I chose it because I have to use these names for the picker library. I thought it would be tidier and more self documenting if they were grouped to the picker since I don't think these styles need to be re-used anywhere.

Maybe I could flatten them out with prefixed names and re-name them before passing them to the component if you don't like this approach.

marcaaron replied:
Seems Ok to me. We are not likely going to re-use these outside of this context so having them together here seems fine for now.

marcaaron commented on src/pages/SettingsPage.js:
```diff
@@ -35,18 +39,40 @@ const propTypes = {
     myPersonalDetails: {},
     network: null,
     session: {},
+    priorityMode: 'default',
```
https://github.com/Expensify/Expensify.cash/blob/20f91178454efb30bdc1c2cc84d5ae3df56dfb2b/src/CONST.js#L35-L38

Julesssss replied:
+1

marcaaron commented on src/pages/SettingsPage.js:
```diff
@@ -35,18 +39,40 @@ const propTypes = {
+
+const priorityModes = {
+    default: {
+        value: 'default',
```
Please use the constant

marcaaron commented on src/pages/SettingsPage.js:
```diff
@@ -35,18 +39,40 @@ const propTypes = {
+        description: 'This will display all chats by default, sorted by most recent, with pinned items at the top',
+    },
+    gsd: {
+        value: 'gsd',
```
Please use the constant

marcaaron commented on src/pages/SettingsPage.js:
```diff
@@ -82,9 +108,9 @@ const SettingsPage = ({
-                    </Text>
+                        <Text style={[styles.settingsLoginName, styles.mt1]} numberOfLines={1}>
+                            {Str.removeSMSDomain(session.email)}
+                        </Text>
```
This change does not seem necessary.

marcaaron commented on src/pages/SettingsPage.js:
```diff
@@ -94,6 +120,29 @@ const SettingsPage = ({
+                            />
+                        </View>
+                        <Text style={[styles.textP, styles.colorMuted]}>
+                            {priorityModes[priorityMode]?.description }
```
Sorry, we don't really use optional chaining so this needs to be done an alternate way. Separately, not sure I see the case where this wouldn't have a value or what we're trying to avoid. If there was such a case we'd then still be returning a `Text` component with no contents?

anthony-hull replied:
Sorry I added this when testing as the app wouldn't recover from a crash when the cause was persisted by ONXY giving an invalid priority mode in props and therefore making an invalid reference here. It shouldn't be needed in prod code.
good catch thank you!

marcaaron commented on src/pages/SettingsPage.js:
```diff
@@ -94,6 +120,29 @@ const SettingsPage = ({
+                                onValueChange={updatePriorityMode}
+                                items={Object.values(priorityModes)}
+                                style={{
+                                    ...styles.picker,
```
Can we just make this `style={styles.picker}`?

anthony-hull replied:
yes. Thanks, I was merging extra properties there for ease while developing. I missed it on the cleanup 👍

marcaaron commented on src/pages/SettingsPage.js:
```diff
@@ -94,6 +120,29 @@ const SettingsPage = ({
+                                    ...styles.picker,
+                                }}
+                                useNativeAndroidPickerStyle={false}
+                                placeholder={{}}
```
Is there a reason this is being passed an empty object?

anthony-hull replied:
Yes. It is to make sure there is no placeholder. There is some default placeholder that is present if you don't pass the empty object.

marcaaron replied:
Hmm, but it always has a `value` so why does it need an explicitly empty placeholder?

marcaaron replied:
If there is some reason it would be good to note in a comment so we can avoid someone accidentally removing this (if not then we should be able to remove this).

marcaaron replied:
Ah I see the problem. The placeholder is a _selectable option_ in the list and not an actual placeholder. Yeah, would be good to add a comment for that. It's not super obvious that would happen.

anthony-hull replied:
I've added a comment. It's outside the React component tag because the linter doesn't like empty lines inside but insists on empty lines above comments. this seemed the cleanest method.

marcaaron commented on src/libs/OptionsListUtils.js:
```diff
@@ -159,17 +160,22 @@ function getOptions(reports, personalDetails, draftComments, activeReportID, {
 
     const reportMapForLogins = {};
-    const orderedReports = lodashOrderBy(reports, [
+    let orderedReports = lodashOrderBy(reports, [
```
It seems kind of wasteful to reassign this and have to filter reports twice (they can be some huge number)

Should be possible to do something like this and avoid the double sort?

```js
const sortProperty = [sortByAlphaAsc ? 'reportName' : blah blah];
const sortDirection = [sortByAlphaAsc ? 'asc' : 'desc'];
const orderedReports = lodashOrderBy(reports, sortProperty, sortDirection)
```

anthony-hull replied:
👍  I like this idea.

marcaaron reviewed (changes requested):
Overall looks good so far but I've got a few comments. Also going to call on @shawnborton to double check the styles to make sure everything is above board.

shawnborton commented on src/pages/SettingsPage.js:
```diff
@@ -94,6 +120,29 @@ const SettingsPage = ({
                         </Text>
                     </TouchableOpacity>
+                    <View style={[styles.settingsPageBody, styles.mt6]}>
+                        <Text style={[styles.settingsPageHeading, styles.mt2, styles.mb4]} numberOfLines={1}>
```
For this one, I apologize for the late change but could we actually just do `styles.textLabel, styles.textStrong` instead of adding a new style for `settingsPageHeading`? Also, I think we could make the bottom margin here a bit smaller, maybe try `styles.mb2` and see how that feels.

shawnborton replied:
And thinking on this some more, maybe we can just make this label say "Chat List Mode"? And then kill the text that immediately follows since we have small helper text that appears below the select menu.

shawnborton commented on src/pages/SettingsPage.js:
```diff
@@ -94,6 +120,29 @@ const SettingsPage = ({
+                                Icon={() => <Icon src={DownArrow} />}
+                            />
+                        </View>
+                        <Text style={[styles.textP, styles.colorMuted]}>
```
Could we make the text here use `styles.textLabel` so it's a bit smaller?

shawnborton commented:
Overall looking pretty good - I left a couple of comments but I have a few more:

- Could we move the "Sign Out" button to be below the GSD mode form?
- Can you make sure the text found inside of the GSD select menu uses the correct brand font (`fontFamily: fontFamily.GTA, fontSize: variables.fontSizeNormal`), as well as `themeColors.text`?

shawnborton commented on src/pages/SettingsPage.js:
```diff
@@ -94,6 +120,29 @@ const SettingsPage = ({
+                            Chats
+                        </Text>
+                        <Text style={[styles.textP]}>
+                            How should we display chats on your home screen?
```
See my comment above, but I think we could kill this.

anthony-hull replied:
> > Looking good. If we change the name to 'Chat List Mode', then perhaps we should update named references of 'priority mode' also? But this is minor and def not a blocker
> 
> Ah yeah, good point. I think we should just use "Priority Mode" as the form label then.

I've changed this to:
Priority Mode
Is that what you meant?

Julesssss commented on src/pages/home/sidebar/SidebarLinks.js:
```diff
@@ -74,6 +77,7 @@ const defaultProps = {
     },
     network: null,
     currentlyViewedReportID: '',
+    priorityMode: 'default',
```
Use constant here, too

Julesssss commented on tests/unit/OptionsListUtilsTest.js:
```diff
@@ -304,9 +312,9 @@ describe('OptionsListUtils', () => {
+    it('getSidebarOptions() with default priority mode', () => {
         // When we call getSidebarOptions() with no search value
-        const results = OptionsListUtils.getSidebarOptions(REPORTS, PERSONAL_DETAILS, {}, 0);
+        const results = OptionsListUtils.getSidebarOptions(REPORTS, PERSONAL_DETAILS, {}, 0, 'default');
```
Const will need to be updated here, too

Julesssss reviewed (approved):
Looking good. If we change the name to 'Chat List Mode', then perhaps we should update named references of 'priority mode' also? But this is minor and def not a blocker

I think we should have an empty state for GSD mode and hide the chat that remains when you switch to GDS mode, but also not a blocker

Julesssss reviewed (changes requested):
Actually, one small issue is that we always show 'Most Recent' in the priority selection dropdown when opening the Settings page. This should show GSD if you are in currently GSD mode.

anthony-hull commented:
Maybe something like
all done, you smashed it! 👊
Would be cool; In the empty chat list when the last chat is read. But you'd have to have some way of dismissing the last chat as it wont filter until it's not longer the the active chat.

Another thought on this new user flow is you might read a message but not want to respond yet or want to switch chats mid flow. In these cases you might not want these chats to be filtered when you're trying to "GSD"
Possibly leaving in chats with drafts might be a good idea?


Get Outlook for iOS<https://aka.ms/o0ukef>
________________________________
From: Jules Rosser <notifications@github.com>
Sent: Wednesday, February 24, 2021 2:07:23 PM
To: Expensify/Expensify.cash <Expensify.cash@noreply.github.com>
Cc: Anthony Hull <anthony.hull@hotmail.co.uk>; Author <author@noreply.github.com>
Subject: Re: [Expensify/Expensify.cash] Anthony 1452 add gsd mode to sidebar (#1553)


@Julesssss approved this pull request.

Looking good. If we change the name to 'Chat List Mode', then perhaps we should update named references of 'priority mode' also? But this is minor and def not a blocker

I think we should have an empty state for GSD mode and hide the chat that remains when you switch to GDS mode, but also not a blocker

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub<https://emea01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2FExpensify%2FExpensify.cash%2Fpull%2F1553%23pullrequestreview-597513913&data=04%7C01%7C%7C53d95b304e3a47a1d7d508d8d8cd82ee%7C84df9e7fe9f640afb435aaaaaaaaaaaa%7C1%7C0%7C637497724454557903%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C1000&sdata=%2B1XWUMoADwOyKKPeKjgxYBtL99y3HS%2BfBYM2SSYrm%2FM%3D&reserved=0>, or unsubscribe<https://emea01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fnotifications%2Funsubscribe-auth%2FALH7R5SJPJXU6SAQGINQ4HLTAUBZXANCNFSM4YDQPA2A&data=04%7C01%7C%7C53d95b304e3a47a1d7d508d8d8cd82ee%7C84df9e7fe9f640afb435aaaaaaaaaaaa%7C1%7C0%7C637497724454562882%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C1000&sdata=5hZ6FLl%2B5jpBp34Zs4lmiTUqS6VjzG7DqxzItG8ZIFc%3D&reserved=0>.

shawnborton commented:
> Looking good. If we change the name to 'Chat List Mode', then perhaps we should update named references of 'priority mode' also? But this is minor and def not a blocker

Ah yeah, good point. I think we should just use "Priority Mode" as the form label then.

shawnborton commented:
In terms of the empty state, I like the idea, but I don't think we need to do that as part of this PR.

marcaaron commented:
>I don't think we need to do that as part of this PR

Agree, these are nice ideas @anthony-hull, but we can just limit the scope what we set out to do in the issue.

anthony-hull commented:
Agreed. I don’t know where else to put suggestions if I have them

Get Outlook for iOS<https://aka.ms/o0ukef>
________________________________
From: Marc Glasser <notifications@github.com>
Sent: Thursday, February 25, 2021 8:48:34 PM
To: Expensify/Expensify.cash <Expensify.cash@noreply.github.com>
Cc: Anthony Hull <anthony.hull@hotmail.co.uk>; Mention <mention@noreply.github.com>
Subject: Re: [Expensify/Expensify.cash] Anthony 1452 add gsd mode to sidebar (#1553)


I don't think we need to do that as part of this PR

Agree, these are nice ideas @anthony-hull<https://emea01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fanthony-hull&data=04%7C01%7C%7C1aef2d78323f4fbbf37f08d8d9ceb83b%7C84df9e7fe9f640afb435aaaaaaaaaaaa%7C1%7C0%7C637498829158139899%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C1000&sdata=d5XJWcA6Sf85hSAJPVIZKgjkxntKScDDuF4a%2BnzdO4E%3D&reserved=0>, but we can just limit the scope what we set out to do in the issue.

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub<https://emea01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2FExpensify%2FExpensify.cash%2Fpull%2F1553%23issuecomment-786193289&data=04%7C01%7C%7C1aef2d78323f4fbbf37f08d8d9ceb83b%7C84df9e7fe9f640afb435aaaaaaaaaaaa%7C1%7C0%7C637498829158149855%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C1000&sdata=cqEpB6Jnrk5Fc7vzvZ6%2FopbsjXyFXNLObDgMSPkOyJM%3D&reserved=0>, or unsubscribe<https://emea01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fnotifications%2Funsubscribe-auth%2FALH7R5V5EUR2I67TJNSVIULTA2ZSFANCNFSM4YDQPA2A&data=04%7C01%7C%7C1aef2d78323f4fbbf37f08d8d9ceb83b%7C84df9e7fe9f640afb435aaaaaaaaaaaa%7C1%7C0%7C637498829158149855%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C1000&sdata=SKduP%2BNRLj5jeiEhxSKqk%2BYG9z%2F8X0eSjz3IhEVN2BY%3D&reserved=0>.

anthony-hull committed:
back of front end PR review changes

anthony-hull commented:
> Actually, one small issue is that we always show 'Most Recent' in the priority selection dropdown when opening the Settings page. This should show GSD if you are in currently GSD mode.

Good catch, thank you.

anthony-hull committed:
styling and layout changes

anthony-hull commented:
![GSD sidebar visual tweaks](https://user-images.githubusercontent.com/47184118/109386924-f3f25e00-78f5-11eb-98d3-f5634c8cda58.png)
Please see attachment for the result of the visual changes.
I have broken the changes down into two commits. First for the back of front end. The second for the visual changes.

anthony-hull committed:
added more references to CONST, remove unnecessary change

shawnborton commented on src/pages/SettingsPage.js:
```diff
@@ -86,6 +113,25 @@ const SettingsPage = ({
+                        <Text style={[styles.textLabel, styles.textStrong, styles.mt2, styles.mb2]} numberOfLines={1}>
+                            Priority Mode
+                        </Text>
+                        <View style={[styles.mt2, styles.mb2]}>
```
Can we get rid of the `mt2` here?

anthony-hull replied:
![image](https://user-images.githubusercontent.com/47184118/109482597-1ad79e00-7a76-11eb-9961-4efe20eb68ae.png)

shawnborton replied:
This looks better, thanks!

shawnborton commented on src/pages/SettingsPage.js:
```diff
@@ -86,6 +113,25 @@ const SettingsPage = ({
                         {Str.removeSMSDomain(session.email)}
                     </Text>
                     )}
+                    <View style={[styles.settingsPageBody, styles.mt4]}>
```
I think it would be nice to add some margin bottom to this view so that there is more separation between this form field block and the sign out button. Perhaps just try using `mb5` but we might need to make a larger margin bottom style if the space isn't quite large enough.

anthony-hull replied:
mb5:
![image](https://user-images.githubusercontent.com/47184118/109482939-86ba0680-7a76-11eb-9d1e-c65ed7eaccd6.png)

mb6:
![image](https://user-images.githubusercontent.com/47184118/109482890-7b66db00-7a76-11eb-9dad-71f785eea59e.png)

anthony-hull replied:
mb6 already exists in the spacing.js 👍 
let me know which one you want and I'll make the commit

shawnborton replied:
Let's do mb6, thanks!

shawnborton commented on src/pages/SettingsPage.js:
```diff
@@ -86,6 +113,25 @@ const SettingsPage = ({
                     </Text>
                     )}
+                    <View style={[styles.settingsPageBody, styles.mt4]}>
+                        <Text style={[styles.textLabel, styles.textStrong, styles.mt2, styles.mb2]} numberOfLines={1}>
```
Okay apologies for the late feedback here, but I actually think we also want the label color to use `themeColors.heading`

So that being said, seems like we have a lot of styles we're chaining together here and I think we will be reusing this in many other places. So I think it makes sense to make a new style like `styles.formLabel` that has:

```
color: themeColors.heading,
fontFamily: fontFamily.GTA_BOLD,
fontSize: variables.fontSizeLabel,
fontWeight: fontWeightBold,
```

Thoughts?

anthony-hull replied:
I think this is a good idea. I can see this being re-used once the menus are expanded with more options in issue 1493
and maybe for the user details labels in the new profile view on 1492.

anthony-hull replied:
We already have a styles.formLabel implemented:
```
    formLabel: {
        color: themeColors.text,
        fontSize: variables.fontSizeLabel,
        lineHeight: 18,
        marginBottom: 8,
    },
```
It looks like this in this situation:
![image](https://user-images.githubusercontent.com/47184118/109486666-15c91d80-7a7b-11eb-891b-f58ab5e94881.png)

So I've made a new one named settingsLabel. Whoever said naming things is hard was right :)
In the other label style the margin was in the style. I think it's likely you're going to always want a margin below the label. We could add `marginBottom: 8`
What do you think?
Here is the result of the changes:
![image](https://user-images.githubusercontent.com/47184118/109487145-bd465000-7a7b-11eb-9cde-258141c6f4f2.png)

shawnborton replied:
Ah, I say we just reuse the existing `formLabel` style and just add `fontFamily: fontFamily.GTA_BOLD, fontWeight: fontWeightBold,` to it. This way we only have one global formLabel style.

anthony-hull replied:
do you want the text colour for the label style to be text or heading? It's currently text.

Additionally form label is implemented in other places such as changing the login page to:
![loginscreen change](https://user-images.githubusercontent.com/47184118/109536433-a0c70980-7ab5-11eb-890d-a25edded934c.png)
compared to before:
![loginscreen change](https://user-images.githubusercontent.com/47184118/109536570-cc49f400-7ab5-11eb-8e9c-1722dea131ca.png)
All the usages are as follows:
![image](https://user-images.githubusercontent.com/47184118/109536674-ec79b300-7ab5-11eb-9882-360a0c91a7a4.png)

shawnborton replied:
Yup, let's use `themeColors.heading` and it is okay if this updates everywhere. Thanks!

anthony-hull replied:
ok I've pushed these changes to the branch 👍

shawnborton replied:
Awesome, thanks Anthony!

anthony-hull committed:
update formLabel styling and apply to Priority mode label

Julesssss reviewed (dismissed):
Changes look good, thanks for fixing the 'use current sort value' issue. Just need to resolve a few conflicts.

marcaaron commented on src/libs/OptionsListUtils.js:
```diff
@@ -159,17 +161,20 @@ function getOptions(reports, personalDetails, draftComments, activeReportID, {
-            ? 'lastMessageTimestamp'
-            : 'lastVisitedTimestamp',
-    ], ['desc']);
+    let iteratees = sortByLastMessageTimestamp
```
NAB, (not a blocker) `sortProperty` rolls off the tongue a bit nicer than `iteratees`

marcaaron commented on src/libs/OptionsListUtils.js:
```diff
@@ -159,17 +161,20 @@ function getOptions(reports, personalDetails, draftComments, activeReportID, {
+    let iteratees = sortByLastMessageTimestamp
+        ? ['lastMessageTimestamp']
+        : ['lastVisitedTimestamp'];
+    if (sortByAlphaAsc) { iteratees = ['reportName']; }
```
Can we expand this into separate lines please? It's uncommon to see a single line if statement in our codebase.

marcaaron commented on src/libs/OptionsListUtils.js:
```diff
@@ -368,18 +375,32 @@ function getNewGroupOptions(
+    let sideBarOptions = {
+        prioritizePinnedReports: true,
+        hideReadReports: false,
+        sortByAlphaAsc: false,
```
NAB, `hideReadReports` and `sortByAlphaAsc` are `false` by default so we don't need to be explicit here.

marcaaron reviewed (changes requested):
This is looking great and testing well on web. I have just one or two small comments and will move on to testing other platforms.

marcaaron commented:
Tests well on all platforms. Also still has conflicts.

anthony-hull committed:
minor PR comment tweaks
renaming variables, removing unnecessary explicit assignments
comment to explain placeholder empty object

anthony-hull committed:
Merge branch 'master' into anthony-1452-add-GSD-mode-to-sidebar

Jag96 commented:
@anthony-hull it looks like there is a bug occurring with this PR:
1. With `Priority Mode` set to `Most Recent`, pin some chats
2. Switch `Priority Mode` to `GSD`, see pinned chats disappear from the list of chats

Can you create a follow-up PR to fix this bug?
